### PR TITLE
Add a blog post about the current state of Flink on Docker

### PR DESCRIPTION
**Pull request by rmetzger**
This blog post hasn't been proofread by anybody beforehand, so this is mostly for collecting early feedback.
I'll cleanup the commit history and date in the blog post later.

### CODE REVIEW

File: _posts/2020-08-20-flink-docker.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+authors:
+- rmetzger:
+  name: "Robert Metzger"
+  twitter: rmetzger_
+categories: news
+
+excerpt: This blog post gives an update on the recent developments of Flink's support for Docker.
+---
+
+The Flink community recently put some effort into upgrading the Docker experience for our users. The goal was to reduce confusion and improve usability. With over 50 million downloads from Docker Hub, the Flink docker images are a very popular deployment option.
```
Comment (sjwiesman) [inline _posts/2020-08-20-flink-docker.md]: quick drive-by suggestion

```suggestion
With over 50 million downloads from Docker Hub, the Flink docker images are a very popular deployment option.
The Flink community recently put some effort into upgrading the Docker experience for our users with the goal to reduce confusion and improve usability.
```

Comment (azagrebin) [inline _posts/2020-08-20-flink-docker.md]: nit:
I am not a native speaker.
`upgrading experience` sounds somewhat strange for me,
maybe because I've encountered only `improving experience` so far

File: _posts/2020-08-20-flink-docker.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+Let's quickly break down the recent improvements:
+
+- Reduce confusion: Flink used to have 2 Dockerfiles and a 3rd file maintained outside of the official repository — all with different features and varying stability. Now, we have one central place for all images: [apache/flink-docker](https://github.com/apache/flink-docker).
+
+  Here, we keep all the Dockerfiles for the different releases. Check out the [detailed readme](https://github.com/apache/flink-docker/blob/master/README.md) of that repository for further explanation on the different branches, as well as the [Flink Improvement Proposal (FLIP-111)](https://cwiki.apache.org/confluence/display/FLINK/FLIP-111%3A+Docker+image+unification) that contains the detailed planning.
+
+  The apache/flink-docker repository also seeds the o[fficial Flink image on Docker Hub](https://hub.docker.com/_/flink).
```
Comment (azagrebin) [inline _posts/2020-08-20-flink-docker.md]:
```suggestion
  The `apache/flink-docker` repository also seeds the [official Flink image on Docker Hub](https://hub.docker.com/_/flink).
```

Comment (azagrebin) [reply _posts/2020-08-20-flink-docker.md]: or some other highlighting for `apache/flink-docker`

File: _posts/2020-08-20-flink-docker.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+- Improve Usability: The Dockerfiles are used for various purposes: [Native docker deployments](https://ci.apache.org/projects/flink/flink-docs-master/ops/deployment/docker.html), [Flink on Kubernetes](https://ci.apache.org/projects/flink/flink-docs-master/ops/deployment/native_kubernetes.html), the (unofficial) [Flink helm example](https://github.com/docker-flink/examples) and the project's [internal end to end tests](https://github.com/apache/flink/tree/master/flink-end-to-end-tests). With one unified image, all these consumers of the images benefit from the same set of features, documentation and testing. 
+
+  The new images support passing configuration variables via a `FLINK_PROPERTIES` environment variable. Users can enable default plugins with the `ENABLE_BUILT_IN_PLUGINS`environment variable. The images also allow loading custom jar paths and configuration files.
```
Comment (azagrebin) [inline _posts/2020-08-20-flink-docker.md]: It may be helpful to turn the phrases for key features into links to the corresponding doc chapters (e.g. FLINK_PROPERTIES, ENABLE_BUILT_IN_PLUGINS, custom jar paths and configuration files). I know we already have a general link but I believe more links do not hurt.

File: _posts/2020-08-20-flink-docker.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+Looking into the future, there are already some interesting potential improvements lined up: 
+
+- [Java 11 Docker images](https://issues.apache.org/jira/browse/FLINK-16260) (already completed)
+- [Use vanilla docker-entrypoint with flink-kubernetes](https://issues.apache.org/jira/browse/FLINK-15793) (in progress)
+- [History server support](https://issues.apache.org/jira/browse/FLINK-17167)
+- [Support for OpenShift](https://issues.apache.org/jira/browse/FLINK-15587)
+
+## How do I get started?
+
+This is a short tutorial on how to start a Flink Session Cluster with docker.
```
Comment (azagrebin) [inline _posts/2020-08-20-flink-docker.md]: again I would either make `a Flink Session Cluster with docker` clickable into the docs or add `see also details in docs link`.

File: _posts/2020-08-20-flink-docker.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+cd flink-1.11.1
+
+# 2: Start the Flink job
+./bin/flink run ./examples/streaming/TopSpeedWindowing.jar
+```
+
+<center>
+<img src="{{ site.baseurl }}/img/blog/flink-docker/flink-docker.gif" width="882px" height="730px" alt="Demo video"/>
+</center>
+
```
Comment (azagrebin) [inline _posts/2020-08-20-flink-docker.md:90]: I would also briefly mention that one could also use docs link to deploy the job mode, just to denote that it is also possible and we envisioned it.

**Review by azagrebin — COMMENTED** [review]
Thanks a lot @rmetzger 
I think the blog post looks really good. It is not overloaded with details and gives a flavour of how docker can be used with Flink. This plus docs links should be a good entry point for users.
I left only some small comments.

File: _posts/2020-08-20-flink-docker.md (hunk 1 above)
Comment (rmetzger) [reply _posts/2020-08-20-flink-docker.md]: Damn, me trying to sound eloquent often fires back :)

Comment (rmetzger) [reply _posts/2020-08-20-flink-docker.md]: Seth didn't correct the word in his review

Comment (rmetzger) [reply _posts/2020-08-20-flink-docker.md]: ... elevating the Docker experience to a new level ... ?

**Comment by rmetzger** [comment]
Thanks a lot for your reviews! I pushed a new version. Unless there's any more feedback, I'll merge it later today.

File: _posts/2020-08-20-flink-docker.md (hunk 1 above)
Comment (rmetzger) [reply _posts/2020-08-20-flink-docker.md]: okay, I'll use "improve"
